File: Java/TwoSum/TwoSum.java
import java.util.Arrays;
import java.util.HashMap;
public class TwoSum
{
    public int[] twoSum(int[] nums, int target)
    {
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i=0; i<nums.length; i++)
        {
            int complement = target-nums[i];
            if(map.containsKey(complement))
            {
                return new int[] {map.get(complement), i};
            }
            else
            {
                map.put(nums[i], i);
            }
        }

        throw new IllegalArgumentException("No two sum solution.");
    }

    public static void main(String[] args)
    {
        TwoSum twosum = new TwoSum();
        int[] input = {2,3,4,6,3,6,4};
        int target = 8;
        int[] ans =  twosum.twoSum(input, target);
        System.out.println(Arrays.toString(ans));
    }
}

File: Python/TwoSum/two_sum.py
def two_sum(nums, target):
    hash_map = {}
    for i, v in enumerate(nums):
        if target-v in hash_map:
            return [hash_map[target-v], i]
        else:
            hash_map[v] = i
    raise ValueError("No two sum match to target.")

def main():
    nums = [2,3,4,6,3,6,4]
    target = 8
    print(two_sum(nums, target))

if __name__=="__main__":
    main()